### PR TITLE
Refactoring to replace inheritance with composable functions

### PR DESCRIPTION
**Pull request by celikmus**
Being replaced with: https://github.com/AdaptiveConsulting/ReactiveTraderCloud/pull/627

I tidied up appBootstrapper and services to move away from inheritance from ServiceBase and DisposableBase... We don't really need inheritance for those, and the code flows easier as functions. As ServiceBase is not used anymore, I have removed it altogether.

While I removed inheritance, we still use ServiceClient by instantiating in factory functions. This means I haven't altered any Rx behaviour, in fact no behaviour at all. I have fixed the impacted testcases, while there are some existing failing tests for ServiceClient... I am planning to tackle with ServiceClient next, and I will tackle those there...

I used prettier on the files that I worked on... I have also taken out some tsLint rules; which reduced majority of the pre-existing tsLint errors.

### CODE REVIEW

**Comment by tdeheurles** [comment]
good for deployment part

File: src/client/src/services/compositeStatusService.ts [hunk 1]
```diff
@@ -1,100 +1,86 @@
...
-    this.pricingService = pricingService
-    this.referenceDataService = referenceDataService
-    this.blotterService = blotterService
-    this.executionService = executionService
-    this.analyticsService = analyticsService
-    this.serviceStatusStream = this.createServiceStatusStream()
-    this.currentServiceStatusLookup = new ServiceStatusLookup()
+): Object {
+  const disposables = new Subscription()
+  const addDisposable = disposable => {
```
Comment (fdore) [inline src/client/src/services/compositeStatusService.ts]: we are not using esp-js anymore, so we dont need that anymore.

File: src/client/src/system/service/connection.ts [hunk 2]
```diff
@@ -37,7 +46,9 @@ export default class Connection extends DisposableBase {
     this.autoDisconnectDisposable = new SerialSubscription()
     this.connectionUrl = ''
     this.connectionType = ConnectionType.Unknown
-    this.addDisposable(this.autoDisconnectDisposable)
+    const prevProto = Object.getPrototypeOf(this.autoDisconnectDisposable)
+    prevProto.dispose = prevProto.unsubscribe
```
Comment (fdore) [inline src/client/src/system/service/connection.ts]: was it because of esp-js?

**Comment by silviuadaptive** [comment]
Replaced with https://github.com/AdaptiveConsulting/ReactiveTraderCloud/pull/627

**Pull request closed without merging**